SpatialGDK/Build/Scripts/DeploymentLauncher.bat

```diff
@@ -0,0 +1,14 @@
+@echo off
+
+set DEPLOYMENT_LAUNCHER_EXE_PATH="%~dp0..\..\..\..\..\Plugins\UnrealGDK\SpatialGDK\Binaries\ThirdParty\Improbable\Programs\DeploymentLauncher\DeploymentLauncher.exe"
+
+if not exist %DEPLOYMENT_LAUNCHER_EXE_PATH% (
+	echo Error: Deployment launcher executable not found! Please run Setup.bat in your UnrealGDK root to generate it.
+	exit /b 1
+)
+
+%DEPLOYMENT_LAUNCHER_EXE_PATH% %*
+
+pause

```

Comment:
Do we need to pause?

Reply:
The GUI launches the script and if the script fails it would just close and not give the chance for the user to see what's wrong - I suppose the pause was added so that it doesn't automatically close.

Reply:
Yup, pause was added for this reason

Reply:
I don't think the pause is working as intended btw.

Reply:
How so? Where are you seeing this, when calling the deployment launcher script from the Unreal Editor or when directly calling the script from the command line? The latter seems to pause for me

Reply:
From the editor, makes it hard to see what the issue was if it doesn't succeed.

Reply:
That shouldn't happen, thanks. We will take a look

Reply:
The reason why it's not happening actually is because the Editor runs the executable rather than the .bat file - https://github.com/spatialos/UnrealGDK/blob/643f2f1d6075b232b28d4353f2af3644d5613937/SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h#L443

Reply:
Can we either run the bat file or make the `exe` wait for a key before closing?

Reply:
Seems like that still doesn't work and the pause is not respected. After a few tries a potential solution is to launch the script using `system(<cmd>)`, which seems to respect the pause, but might be a little bit unconventional 👀. What do you think?